daemon/openrazer_daemon/hardware/device_base.py

```diff
@@ -104,13 +134,371 @@ def __init__(self, device_path, device_number, config, testing=False, additional
+            ('razer.device.misc', 'getCurrentScrollEffectSpeed', self.get_current_scroll_effect_speed, None, 'i'),
+        }
+
+        left_effect_methods = {
+            ('razer.device.misc', 'getCurrentLeftEffect', self.get_current_left_effect, None, 's'),
+            ('razer.device.misc', 'getCurrentLeftEffectColors', self.get_current_left_effect_colors, None, 'ay'),
+            ('razer.device.misc', 'getCurrentLeftEffectSpeed', self.get_current_left_effect_speed, None, 'i'),
+        }
+
+        right_effect_methods = {
+            ('razer.device.misc', 'getCurrentRightEffect', self.get_current_right_effect, None, 's'),
+            ('razer.device.misc', 'getCurrentRightEffectColors', self.get_current_right_effect_colors, None, 'ay'),
+            ('razer.device.misc', 'getCurrentRightEffectSpeed', self.get_current_right_effect_speed, None, 'i'),
+        }
+
         for m in methods:
             self.logger.debug("Adding {}.{} method to DBus".format(m[0], m[1]))
             self.add_dbus_method(m[0], m[1], m[2], in_signature=m[3], out_signature=m[4])
 
+        if 'set_static_effect' in self.METHODS:
+            self.has_normal_effects = True
+            for m in effect_methods:
+                self.logger.debug("Adding {}.{} method to DBus".format(m[0], m[1]))
+                self.add_dbus_method(m[0], m[1], m[2], in_signature=m[3], out_signature=m[4])
+
+        if 'set_logo_static' in self.METHODS or 'set_logo_static_naga_hex_v2' in self.METHODS:
+            self.has_logo_effects = True
+            for m in logo_effect_methods:
+                self.logger.debug("Adding {}.{} method to DBus".format(m[0], m[1]))
+                self.add_dbus_method(m[0], m[1], m[2], in_signature=m[3], out_signature=m[4])
+
+        if 'set_scroll_static' in self.METHODS or 'set_logo_static_naga_hex_v2' in self.METHODS:
+            self.has_scroll_effects = True
+            for m in scroll_effect_methods:
+                self.logger.debug("Adding {}.{} method to DBus".format(m[0], m[1]))
+                self.add_dbus_method(m[0], m[1], m[2], in_signature=m[3], out_signature=m[4])
+
+        if 'set_left_static' in self.METHODS:
+            self.has_left_effects = True
+            for m in left_effect_methods:
+                self.logger.debug("Adding {}.{} method to DBus".format(m[0], m[1]))
+                self.add_dbus_method(m[0], m[1], m[2], in_signature=m[3], out_signature=m[4])
+
+        if 'set_right_static' in self.METHODS:
+            self.has_right_effects = True
+            for m in right_effect_methods:
+                self.logger.debug("Adding {}.{} method to DBus".format(m[0], m[1]))
+                self.add_dbus_method(m[0], m[1], m[2], in_signature=m[3], out_signature=m[4])
+
         # Load additional DBus methods
         self.load_methods()
 
+        # load last effect
+        if self.has_normal_effects:
+            if self.config.has_section(self._serial):
```

Comment:
Some old devices use an temporary generated serial (see driver source), that should be maybe changed to always start with e.g. ABC000000 and then count up per devices plugged in so we can have some kind of certainty that the serial will exist again

Reply:
Done. Serial-less devices will be stored by name instead.